spring-integration-file/src/main/java/org/springframework/integration/file/support/FileUtils.java

```diff
@@ -41,9 +42,9 @@
 	 */
 	@SuppressWarnings("unchecked")
 	public static <F> F[] purgeUnwantedElements(F[] fileArray, Predicate<F> predicate) {
-		return (F[]) Arrays.stream(fileArray)
+		return Arrays.stream(fileArray)
 				.filter(predicate.negate())
-				.toArray();
+				.toArray(size -> (F[]) Array.newInstance(fileArray[0].getClass(), size));
```

Comment:
I think we need to add Assert about the length in the beginning of the method.
Or just proceed to the `return` when there are no elements in the original array...